modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptNestjsClientCodegen.java

```diff
@@ -0,0 +1,603 @@
+    public static final String MODEL_FILE_SUFFIX = "modelFileSuffix";
+    public static final String FILE_NAMING = "fileNaming";
+    public static final String STRING_ENUMS = "stringEnums";
+    public static final String STRING_ENUMS_DESC = "Generate string enums instead of objects for enum values.";
+
+    protected String nestVersion = "6.0.0";
+    protected String npmRepository = null;
+    protected String serviceSuffix = "Service";
+    protected String serviceFileSuffix = ".service";
+    protected String modelSuffix = "";
+    protected String modelFileSuffix = "";
+    protected String fileNaming = "camelCase";
+    protected Boolean stringEnums = false;
+
+    private boolean taggedUnions = false;
+
+    public TypeScriptNestjsClientCodegen() {
+        super();
+        this.outputFolder = "generated-code/typescript-nestjs";
+
+        supportsMultipleInheritance = true;
+
+        embeddedTemplateDir = templateDir = "typescript-nestjs";
+        modelTemplateFiles.put("model.mustache", ".ts");
+        apiTemplateFiles.put("api.service.mustache", ".ts");
+        languageSpecificPrimitives.add("Blob");
+        typeMapping.put("file", "Blob");
+        apiPackage = "api";
+        modelPackage = "model";
+
+        this.cliOptions.add(new CliOption(NPM_REPOSITORY,
+                "Use this property to set an url your private npmRepo in the package.json"));
+        this.cliOptions.add(CliOption.newBoolean(WITH_INTERFACES,
+                "Setting this property to true will generate interfaces next to the default class implementations.",
+                false));
+        this.cliOptions.add(CliOption.newBoolean(TAGGED_UNIONS,
+                "Use discriminators to create tagged unions instead of extending interfaces.",
+                this.taggedUnions));
+        this.cliOptions.add(CliOption.newBoolean(PROVIDED_IN_ROOT,
+                "Use this property to provide Injectables in root (it is only valid in nestjs version greater or equal to 6.0.0).",
+                false));
+        this.cliOptions.add(new CliOption(NEST_VERSION, "The version of Nestjs.").defaultValue(this.nestVersion));
+        this.cliOptions.add(new CliOption(SERVICE_SUFFIX, "The suffix of the generated service.").defaultValue(this.serviceSuffix));
+        this.cliOptions.add(new CliOption(SERVICE_FILE_SUFFIX, "The suffix of the file of the generated service (service<suffix>.ts).").defaultValue(this.serviceFileSuffix));
+        this.cliOptions.add(new CliOption(MODEL_SUFFIX, "The suffix of the generated model."));
+        this.cliOptions.add(new CliOption(MODEL_FILE_SUFFIX, "The suffix of the file of the generated model (model<suffix>.ts)."));
+        this.cliOptions.add(new CliOption(FILE_NAMING, "Naming convention for the output files: 'camelCase', 'kebab-case'.").defaultValue(this.fileNaming));
+        this.cliOptions.add(new CliOption(STRING_ENUMS, STRING_ENUMS_DESC).defaultValue(String.valueOf(this.stringEnums)));
```

Comment:
Most of these cli options duplicate the ones defined in the parent class. Is there a particular reason to duplicate it here also?

Reply:
I cannot find the duplication, link?